spec/system/posts_spec.rb

```diff
@@ -1,35 +1,59 @@
 require 'rails_helper'
 
 RSpec.describe '投稿機能', type: :system do
-  let(:user) { FactoryBot.create(:user) }
+  let!(:user) { FactoryBot.create(:user) }
+  let!(:other_user) { FactoryBot.create(:user) }
```

Comment:
user, other_userだとパッと読めないことがあるので、変数名でわかり易い名前をつけるとよいケースかもです
```suggestion
  let!(:taro) { FactoryBot.create(:user, name: 'taro') }
  let!(:hanako) { FactoryBot.create(:user, name: 'hanako') }
```